Add Git variables migration documentation (#1641)
OctopusDeploy/docs: docs/projects/version-control/converting/index.md

---
title: Converting projects to Git
description: Converting a project to leverage the configuration as a code feature.
position: 10 
---

Git settings are configured per project and are accessed via the **{{ Settings, Version Control }}** link in the project navigation menu. This page will walk through how to convert a project to Git.

![Version-control configuration UI](version-control-configuration.png "width=500")

## Creating a new version-controlled project

> To get a feel for the config-as-code feature, you may want to create a new project that you can test with before committing to permanently converting an existing project. This project's deployment process, deployment settings, and variables will be stored in a Git repository when configured.

Reviewer: ```suggestion
To get a feel for the config-as-code feature, you may want to create a new project that you can test before committing to permanently converting an existing project. This project's deployment process, deployment settings, and non-sensitive variables will be stored in a Git repository when configured.
```

> To get a feel for the config-as-code feature, you may want to create a new project that you can test with before committing to permanently converting an existing project. This project's deployment process, deployment settings, and variables will be stored in a Git repository when configured.

Reviewer 2: ```suggestion
To get a feel for the config-as-code feature, it is recommended that a new project is created that can be tested before committing to permanently converting an existing project. This project's deployment process, deployment settings, and variables will be stored in a Git repository when configured.
```

Author: I'd like to keep this one as active as possible, and keep it as an optional task (not necessarily a recommended one).

Click the **New Project** button and select **Use version control for this project.**

![adding a project using vcs](add-project-vcs.png)

Once you click the save button, you'll be sent to the version control screen to configure your version control settings. Enter the URL for your Git repository, the name of the default branch, your username and password / personal access token. 

Learn more about [Git credentials in Octopus Deploy](/docs/projects/version-control/config-as-code-reference.md).

Next, add the directory you would like Octopus to store the project configuration. You can connect multiple projects to the same repository if they all use a different sub-directory (e.g. `.octopus/acme` and `.octopus/another-project`).

## Configuring an existing project to use Git

:::warning

> Converting a project to use Git is a one-way change. Once you convert a project Git, you **cannot** converted it back. Please make sure you want to do this, and consider cloning your project to test how it works, so you know what to expect before converting important projects.

Reviewer: ```suggestion
Converting a project to use Git is a one-way change. Once you convert a project Git, you **cannot** convert it back. Please make sure you want to do this, and consider cloning your project to test how it works, so you know what to expect before converting important projects.
```

![configuring version control](configure-version-control.png)

Your project is now configured with Version Control. You can see this change reflected on the Process page, where you can change branches. You can also confirm this in your Git repository. The `.octopus` directory will now be created, and it should contain the following files:

> - _deployment_process.ocl_,
> - _deployment_settings.ocl_,
> - _variables.ocl_,
> - _schema_version.ocl_. 

Reviewer: ```suggestion
- _deployment_process.ocl_
- _deployment_settings.ocl_
- _variables.ocl_
- _schema_version.ocl_
```
